Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py

```diff
@@ -22,27 +21,31 @@
 
 class Client:
     @logger
-    def __init__(self, app_id: str, verify: bool, proxy: bool, base_url: str = BASE_URL):
+    def __init__(self, app_id: str, verify: bool, proxy: bool, base_url: str = BASE_URL, tenant_id: str = None,
+                 enc_key: str = None, client_credentials: bool = False):
         if '@' in app_id:
             app_id, refresh_token = app_id.split('@')
             integration_context = get_integration_context()
             integration_context.update(current_refresh_token=refresh_token)
             set_integration_context(integration_context)
 
-        client_args = {
-            'self_deployed': True,  # We always set the self_deployed key as True because when not using a self
+        self.client_credentials = client_credentials
+        client_args = assign_params(
+            self_deployed=True,  # We always set the self_deployed key as True because when not using a self
             # deployed machine, the DEVICE_CODE flow should behave somewhat like a self deployed
             # flow and most of the same arguments should be set, as we're !not! using OProxy.
-            'auth_id': app_id,
-            'token_retrieval_url': 'https://login.windows.net/organizations/oauth2/v2.0/token',
-            'grant_type': DEVICE_CODE,
-            'base_url': base_url,
-            'verify': verify,
-            'proxy': proxy,
-            'scope': 'offline_access https://security.microsoft.com/mtp/.default',
-            'ok_codes': (200, 201, 202, 204),
-            'resource': 'https://api.security.microsoft.com'
-        }
+            tenant_id=tenant_id,
+            auth_id=app_id,
+            token_retrieval_url='https://login.windows.net/organizations/oauth2/v2.0/token' if not client_credentials else None,
+            grant_type=CLIENT_CREDENTIALS if client_credentials else DEVICE_CODE,
+            base_url=base_url,
+            verify=verify,
+            proxy=proxy,
+            scope='offline_access https://security.microsoft.com/mtp/.default',
+            ok_codes=(200, 201, 202, 204),
+            resource='https://api.security.microsoft.com' if not client_credentials else None,
+            enc_key=enc_key,
+        )
```

Comment:
seperate by relevant authentication flows
```suggestion
        client_args = assign_params(
            base_url=base_url,
            verify=verify,
            proxy=proxy,
            ok_codes=(200, 201, 202, 204),
            scope='offline_access https://security.microsoft.com/mtp/.default',

            self_deployed=True,  # We always set the self_deployed key as True because when not using a self
            # deployed machine, the DEVICE_CODE flow should behave somewhat like a self deployed
            # flow and most of the same arguments should be set, as we're !not! using OProxy.

            grant_type=CLIENT_CREDENTIALS if client_credentials else DEVICE_CODE,
            
            # used for device code flow
            resource='https://api.security.microsoft.com' if not client_credentials else None,
            token_retrieval_url='https://login.windows.net/organizations/oauth2/v2.0/token' if not client_credentials else None,

            # used for client credentials flow
            tenant_id=tenant_id,
            auth_id=app_id,
            enc_key=enc_key,
        )
```